Client/Frontend/Browser/BackForwardListView.swift

```diff
@@ -0,0 +1,196 @@
+            make.height.equalTo(0)
+            make.left.right.bottom.equalTo(self)
+        }
+        
+        if currentRow > 1 {
+            self.tableView.reloadData()
+            let moveToIndexPath = NSIndexPath(forRow: currentRow-2, inSection: 0)
+            self.tableView.reloadRowsAtIndexPaths([moveToIndexPath], withRowAnimation: .None)
+            self.tableView.scrollToRowAtIndexPath(moveToIndexPath, atScrollPosition: UITableViewScrollPosition.Middle, animated: false)
+        }
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        
+        if dismissing {
+            return
+        }
+        
+        let height = min(CGFloat(BackForwardViewUX.RowHeight*listData.count), self.frame.height/2)
+        UIView.animateWithDuration(0.2, animations: {
+            self.tableView.snp_updateConstraints { make in
+                make.height.equalTo(height)
+            }
+            self.backgroundColor = UIColor.init(colorLiteralRed: 0, green: 0, blue: 0, alpha: 0.2)
+            self.layoutIfNeeded()
+        })
+    }
+    
+    // MARK: - Table view
+    
+    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return listData.count
+    }
+    
+    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        var cell: BackForwardTableViewCell? = tableView.dequeueReusableCellWithIdentifier("BackForwardListViewController") as? BackForwardTableViewCell
+        if (cell == nil)
+        {
+            cell = BackForwardTableViewCell(style: UITableViewCellStyle.Default,
+                                            reuseIdentifier: "BackForwardListViewController", isPrivate: isPrivate)
+        }
+        let item = listData[indexPath.item].item
+        
+        if let site = sites[item.URL.absoluteString] {
+            cell?.site = site
+        }
+        else {
+            cell?.site = Site(url: item.initialURL.absoluteString, title: item.title ?? "")
+        }
+        
+        
+        cell?.currentTab = (listData[indexPath.item].type == .Current)
+        cell?.connectingBackwards = (indexPath.item != listData.count-1)
+        cell?.connectingForwards = (indexPath.item != 0)
+        
+        cell?.setNeedsDisplay()
+        
+        return cell!
+    }
+    
+    
+    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        tabManager.selectedTab?.goToBackForwardListItem(listData[indexPath.item].item)
+        dismissWithAnimation()
+    }
+    
+    func tableView(tableView: UITableView, heightForRowAtIndexPath  indexPath: NSIndexPath) -> CGFloat {
+        return CGFloat(BackForwardViewUX.RowHeight);
+    }
+    
+    override func pointInside(point: CGPoint, withEvent event: UIEvent?) -> Bool {
```

Comment:
Is this needed if the user taps outside of the table view? I thought the table view was the entire view? Even if it isn't it might be better to just use a `UITapGestureRecognizer` instead of having to check if a point is within a frame.